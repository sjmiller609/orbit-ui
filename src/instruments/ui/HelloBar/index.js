'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import s from './styles.scss'
import { Row } from 'instruments'

class HelloBar extends React.Component {
  render() {
    const { msg } = this.props
    return (
      <Row className={classnames(s.notice, msg ? s.show : null)}>
        <div className={s.message}>{msg}</div>
      </Row>
    )
  }
}

HelloBar.propTypes = {
  msg: PropTypes.string,
  setMsg: PropTypes.func,
}

export default HelloBar
