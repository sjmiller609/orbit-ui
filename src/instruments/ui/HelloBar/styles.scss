@import 'src/styles/mixins.scss';

.notice {
  font-size: 1.3rem;
  color: $white;
  padding: 0;
  @include font($f500);
  left: 0;
  top: 0;
  width: 100%;
  height: 0;
  background-image: linear-gradient(to right, $orange, $red);
  border-radius: 0;
  box-shadow: 0 -1px 1.2em 0.3em $shadow;

  &.show {
    height: 3rem;
  }

  .message {
    width: 100%;
    text-align: center;
    padding: 0 1.5rem;
  }

  .button {
    font-size: 10px;
    padding: .7em 1.5em .5em;
    margin-left:1rem;
    height: 2.5em; //specifying for 100% children
    color: $white;
    @include shadow;
    background-image: linear-gradient(to right, $purple, $purple);
  }
}