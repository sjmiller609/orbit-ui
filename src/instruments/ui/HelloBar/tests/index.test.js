import React from 'react'
import { shallow } from 'enzyme'
import HelloBar from '../index'

const props = {
  msg: 'Test message',
  to: '/',
  button: 'Click me',
}

describe('HelloBar', () => {
  it('should be defined', () => {
    expect(HelloBar).toBeDefined()
  })

  it('renders correctly', () => {
    const wrapper = shallow(<HelloBar {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should handle no msg', () => {
    const wrapper = shallow(<HelloBar {...props} msg={null} />)
    expect(wrapper).toMatchSnapshot()
  })
})
