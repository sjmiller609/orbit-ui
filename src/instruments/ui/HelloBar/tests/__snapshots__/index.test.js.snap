// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloBar renders correctly 1`] = `
<Row
  className="notice show"
>
  <Row
    className="message"
  >
    Test message
    <Button
      className="button"
      to="/"
    >
      Click me
    </Button>
  </Row>
</Row>
`;

exports[`HelloBar should handle no msg 1`] = `
<Row
  className="notice"
>
  <Row
    className="message"
  />
</Row>
`;
